lib/CFRPageActions.jsm

```diff
@@ -275,6 +276,52 @@ class PageAction {
     }
   }
 
+  async _renderPinTabAnimation() {
+    const footer = this.window.document.getElementById("cfr-notification-footer");
+    const ANIMATION_CONTAINER_ID = "cfr-notification-footer-pintab-animation-container";
+    let animationsEnabled = Services.prefs.getBoolPref(ANIMATIONS_ENABLED_PREF, true);
+    let animationContainer = this.window.document.getElementById(ANIMATION_CONTAINER_ID);
+    if (animationContainer) {
+      animationContainer.toggleAttribute("animate", animationsEnabled);
+      animationContainer.removeAttribute("paused");
+    } else {
+      animationContainer = this.window.document.createElement("vbox");
+      animationContainer.setAttribute("id", ANIMATION_CONTAINER_ID);
+      animationContainer.toggleAttribute("animate", animationsEnabled);
+
+      let spacer = this.window.document.createElement("vbox");
+      spacer.setAttribute("flex", 1);
+      animationContainer.appendChild(spacer);
+
+      let controlsContainer = this.window.document.createElement("hbox");
+      controlsContainer.setAttribute("id", "cfr-notification-footer-animation-controls");
+
+      spacer = this.window.document.createElement("vbox");
+      spacer.setAttribute("flex", 1);
+      controlsContainer.appendChild(spacer);
+
+      let pauseButton = this.window.document.createElement("hbox");
+      pauseButton.setAttribute("id", "cfr-notification-footer-pause-button");
+      pauseButton.addEventListener("click", () => {
+        animationContainer.setAttribute("paused", true);
+      });
+
+      let pauseLabel = this.window.document.createElement("label");
+      pauseLabel.setAttribute("id", "cfr-notification-footer-pause-label");
+      pauseLabel.textContent = await this.getStrings({"string_id": "cfr-doorhanger-pintab-animation-pause"});
+      pauseButton.appendChild(pauseLabel);
+
+      let pauseIcon = this.window.document.createElement("image");
+      pauseIcon.setAttribute("id", "cfr-notification-footer-pause-icon");
+      pauseButton.appendChild(pauseIcon);
```

Comment:
Could we add the icon via CSS as background so we don't create so much markup?

Reply:
it is CSS but I guess maybe it didnt need it's own element. also, a followup since m-c change already shipped.